### ISSUE TITLE
Consider using an AST based parser

### ISSUE THREAD

**Issue by JohnCoates** [angelolloqui/SwiftKotlin#6]
Hi Angel!

Just stumbled upon this project, very cool implementation you've got.  Figured I'd suggest considering an AST based parser since implementing type specific conversions like the array append/add limitation listed in your Readme should be easier than with your current tokenizer. [SwiftLint](https://github.com/realm/SwiftLint) leverages SourceKit for this reason.

**Comment by angelolloqui** [comment]
Hi John,

Yes, indeed I started with SourceKit and a full AST, but it was in the end much more difficult than the tokenizer, so I decided to keep it simple (otherwise I would not have the time to make something functional enough). If you check the differences between Swift and Kotlin, most of it can work with the tokenizer in a much simpler way, although I agree that some special cases (like constructors) can be cumbersome without the AST. 

Anyway, if you want to contribute with an AST based version, I will be happy to incorporate it. 

Thanks for the comments!
